Save Media option saves whole article rather than file

When using the new landing page layout: on android 2.4 a long press on an image will give me the opportunity to "save media"
![savemedia](https://user-images.githubusercontent.com/21675885/50211754-ad9a2f80-0379-11e9-8cb8-9e75b7202a15.jpg)

The problem is that it won't save the image but the .html file.

@macgills  When I try to commit (I bypassed)
```

> Task :custom:processCustomexampleDebugResources FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':custom:processCustomexampleDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     /home/mhutti1/kiwix/kiwix-android/custom/src/main/AndroidManifest.xml:6:3-43:17: AAPT: error: resource string/app_name (aka org.kiwix.kiwixcustomexample:string/app_name) not found.
         
     /home/mhutti1/kiwix/kiwix-android/custom/src/main/AndroidManifest.xml:6:3-43:17: AAPT: error: resource style/AppTheme (aka org.kiwix.kiwixcustomexample:style/AppTheme) not found.
         
     /home/mhutti1/kiwix/kiwix-android/custom/src/main/AndroidManifest.xml:8:5-35:16: AAPT: error: resource string/app_name (aka org.kiwix.kiwixcustomexample:string/app_name) not found.
         
     /home/mhutti1/kiwix/kiwix-android/custom/src/main/AndroidManifest.xml:24:7-30:23: AAPT: error: resource string/app_search_string (aka org.kiwix.kiwixcustomexample:string/app_search_string) not found.
         
     /home/mhutti1/kiwix/kiwix-android/core/build/intermediates/library_manifest/debug/AndroidManifest.xml:33:9-48:20: AAPT: error: resource string/app_name (aka org.kiwix.kiwixcustomexample:string/app_name) not found.
         
     /home/mhutti1/kiwix/kiwix-android/core/build/intermediates/library_manifest/debug/AndroidManifest.xml:33:9-48:20: AAPT: error: resource style/AppTheme.Launcher (aka org.kiwix.kiwixcustomexample:style/AppTheme.Launcher) not found.
         
     /home/mhutti1/kiwix/kiwix-android/core/build/intermediates/library_manifest/debug/AndroidManifest.xml:45:13-47:53: AAPT: error: resource xml/shortcuts (aka org.kiwix.kiwixcustomexample:xml/shortcuts) not found.
         
     /home/mhutti1/kiwix/kiwix-android/core/build/intermediates/library_manifest/debug/AndroidManifest.xml:83:13-85:58: AAPT: error: resource xml/provider_paths (aka org.kiwix.kiwixcustomexample:xml/provider_paths) not found.
         
     /home/mhutti1/.gradle/caches/transforms-2/files-2.1/9d55815a802cb043f369858877d13ab9/jetified-leakcanary-android-core-2.0-alpha-2/AndroidManifest.xml:39:9-44:66: AAPT: error: resource string/leak_canary_display_activity_label (aka org.kiwix.kiwixcustomexample:string/leak_canary_display_activity_label) not found.
         
     /home/mhutti1/.gradle/caches/transforms-2/files-2.1/9d55815a802cb043f369858877d13ab9/jetified-leakcanary-android-core-2.0-alpha-2/AndroidManifest.xml:39:9-44:66: AAPT: error: resource style/leak_canary_LeakCanary.Base (aka org.kiwix.kiwixcustomexample:style/leak_canary_LeakCanary.Base) not found.
         
     /home/mhutti1/.gradle/caches/transforms-2/files-2.1/9d55815a802cb043f369858877d13ab9/jetified-leakcanary-android-core-2.0-alpha-2/AndroidManifest.xml:46:9-59:26: AAPT: error: resource bool/leak_canary_add_launcher_icon (aka org.kiwix.kiwixcustomexample:bool/leak_canary_add_launcher_icon) not found.
         
     /home/mhutti1/.gradle/caches/transforms-2/files-2.1/9d55815a802cb043f369858877d13ab9/jetified-leakcanary-android-core-2.0-alpha-2/AndroidManifest.xml:46:9-59:26: AAPT: error: resource string/leak_canary_display_activity_label (aka org.kiwix.kiwixcustomexample:string/leak_canary_display_activity_label) not found.
         
     /home/mhutti1/.gradle/caches/transforms-2/files-2.1/9d55815a802cb043f369858877d13ab9/jetified-leakcanary-android-core-2.0-alpha-2/AndroidManifest.xml:46:9-59:26: AAPT: error: resource style/leak_canary_LeakCanary.Base (aka org.kiwix.kiwixcustomexample:style/leak_canary_LeakCanary.Base) not found.
         
     /home/mhutti1/.gradle/caches/transforms-2/files-2.1/9d55815a802cb043f369858877d13ab9/jetified-leakcanary-android-core-2.0-alpha-2/AndroidManifest.xml:61:9-67:68: AAPT: error: resource string/leak_canary_storage_permission_activity_label (aka org.kiwix.kiwixcustomexample:string/leak_canary_storage_permission_activity_label) not found.
         
     /home/mhutti1/.gradle/caches/transforms-2/files-2.1/9d55815a802cb043f369858877d13ab9/jetified-leakcanary-android-core-2.0-alpha-2/AndroidManifest.xml:61:9-67:68: AAPT: error: resource style/leak_canary_Theme.Transparent (aka org.kiwix.kiwixcustomexample:style/leak_canary_Theme.Transparent) not found.

@macgills 

```> Task :app:ktlintAndroidTestSourceSetCheck
/home/mhutti1/kiwix/kiwix-android/app/src/androidTest/java/org/kiwix/kiwixmobile/KiwixMockServer.kt:48:1: Exceeded max line length (100) (cannot be auto-corrected)
/home/mhutti1/kiwix/kiwix-android/app/src/androidTest/java/org/kiwix/kiwixmobile/di/components/TestComponent.kt:36:1: Exceeded max line length (100) (cannot be auto-corrected)
"plain" report written to /home/mhutti1/kiwix/kiwix-android/app/build/reports/ktlint/ktlintAndroidTestSourceSetCheck.txt

> Task :app:ktlintAndroidTestSourceSetCheck FAILED

FAILURE: Build failed with an exception.
```

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.java
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.java
@@ -168,6 +168,7 @@ public class KiwixWebView extends VideoEnabledWebView {
       if (url != null || src != null) {
         url = url == null ? src : url;
         url = url.substring(url.lastIndexOf('/') + 1);
+        url = src == null || url.contains(".") ? url : src.substring(src.lastIndexOf('/') + 1);
         url = url.substring(url.indexOf("%3A") + 1);
         int dotIndex = url.lastIndexOf('.');
 

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.java
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.java
@@ -159,6 +159,17 @@ public class KiwixWebView extends VideoEnabledWebView {
   }
 
   static class SaveHandler extends Handler {
+    private String getDecodedFileName(String url, String src) {
+      String fileName = "";
+      if (url != null) {
+        fileName = url.substring(url.lastIndexOf('/') + 1);
+      }
+      // If url is not a valid file name use src if it isn't null
+      if (!fileName.contains(".") && src != null) {
+        fileName = src.substring(src.lastIndexOf('/') + 1);
+      }
+      return fileName.substring(fileName.indexOf("%3A") + 1);
+    }
 
     @Override
     public void handleMessage(Message msg) {
@@ -166,11 +177,8 @@ public class KiwixWebView extends VideoEnabledWebView {
       String src = (String) msg.getData().get("src");
 
       if (url != null || src != null) {
-        url = url == null ? src : url;
-        url = url.substring(url.lastIndexOf('/') + 1);
-        url = src == null || url.contains(".") ? url : src.substring(src.lastIndexOf('/') + 1);
-        url = url.substring(url.indexOf("%3A") + 1);
-        int dotIndex = url.lastIndexOf('.');
+        String fileName = getDecodedFileName(url, src);
+        int dotIndex = fileName.lastIndexOf('.');
 
         File root = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES);
 
@@ -179,10 +187,10 @@ public class KiwixWebView extends VideoEnabledWebView {
           root = CoreApp.getInstance().getExternalMediaDirs()[0];
         }
 
-        File storageDir = new File(root, url);
-        String newUrl = url;
+        File storageDir = new File(root, fileName);
+        String newUrl = fileName;
         for (int i = 2; storageDir.exists(); i++) {
-          newUrl = url.substring(0, dotIndex) + "_" + i + url.substring(dotIndex);
+          newUrl = fileName.substring(0, dotIndex) + "_" + i + fileName.substring(dotIndex);
           storageDir = new File(root, newUrl);
         }
 

